ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/BlockFetch/Server.hs

```diff
@@ -107,17 +108,18 @@ blockFetchServer _tracer chainDB _version registry = senderSide
         -- When we got an iterator, it will stream at least one block since
         -- its bounds are inclusive, so we don't have to check whether the
         -- iterator is empty.
-        Right it -> SendMsgStartBatch $ sendBlocks it
+        Right it -> SendMsgStartBatch $ sendBlocks 0 it
 
-    sendBlocks :: ChainDB.Iterator m blk (WithPoint blk (Serialised blk))
+    sendBlocks :: Int -> ChainDB.Iterator m blk (WithPoint blk (Serialised blk))
                -> m (BlockFetchSendBlocks (Serialised blk) (Point blk) m ())
-    sendBlocks it = do
+    sendBlocks !blockCount it = do
       next <- ChainDB.iteratorNext it
       case next of
         IteratorResult blk     ->
-          return $ SendMsgBlock (withoutPoint blk) (sendBlocks it)
+          return $ SendMsgBlock (withoutPoint blk) (sendBlocks (blockCount + 1) it)
```

Comment:
I think you want to put the `traceWith` here: one event per block sent (no `Int`).

Reply:
Ah ok.  The reason I did the count rather than trace once per message was because I was concerned about the performance implications of emitting an event per block.  If that's not a concern, then it makes sense to do as you say and emit an event per block.